packages/bsky/src/api/app/bsky/feed/searchPosts.ts

```diff
@@ -102,28 +102,43 @@ const hydration = async (
   )
 }
 
-const noBlocks = (inputs: RulesFnInput<Context, Params, Skeleton>) => {
-  const { ctx, skeleton, hydration } = inputs
+const noBlocksOrTagged = (inputs: RulesFnInput<Context, Params, Skeleton>) => {
+  const { ctx, params, skeleton, hydration } = inputs
   skeleton.posts = skeleton.posts.filter((uri) => {
+    const post = hydration.posts?.get(uri)
+    if (!post) return
+
     const creator = creatorFromUri(uri)
-    return !ctx.views.viewerBlockExists(creator, hydration)
+
+    if (ctx.views.viewerBlockExists(creator, hydration)) {
+      return false
+    }
+
+    const { author } = skeleton.parsedQuery
+    const isCuratedSearch = params.sort === 'top'
+    const isPostByViewer = creator === params.hydrateCtx.viewer
+    if (
+      !isPostByViewer &&
+      !params.isModService &&
+      (isCuratedSearch || !author) &&
+      [...ctx.cfg.searchTagsHide].some((t) => post.tags.has(t))
+    ) {
```

Comment:
Nit: could be nice to break these out into different `if` statements so that it's a little more clear what the cases are

Reply:
ac9a7bcb1